torchdata/dataloader2/dataloader2.py

```diff
@@ -49,7 +49,10 @@ def __next__(self) -> T_co:
         if self.iterator_id == self.dataloader.valid_iterator_id:
             self.dataloader._reset_iter = True
             try:
-                return next(self.dataloader._datapipe_iter)  # type: ignore[arg-type]
+                if self.dataloader._paused:
+                    raise StopIteration
+                else:
+                    return next(self.dataloader._datapipe_iter)  # type: ignore[arg-type]
```

Comment:
Thoughts on having DL2 directly check if it is paused or not (versus at the RS level)?

This means providing the (potentially) public API `pause`/`resume`. We can also make them private.

Reply:
It makes sense to have DL2 having those APIs (we can add them as prototype feature raising warning when calling them). And, it also means we need to think about how to add the corresponding APIs to `ReadingService` as well so that DL2 can all them accordingly.

Reply:
Yeah. I am thinking to keep a noop `pause` and `resume` (with warning) to base `ReadingServiceInterface` should be good enough.